TEI.js

```diff
@@ -94,6 +104,50 @@ function replaceFormatting(title) {
 	return titleText;
 }
 
+/**
+ * Append possible rich text html as TEI element (<hi rend="…">)
+ * @param {*} node 
+ * @param {*} html 
+ */
+function appendXML (node, html) {
+	if (!html) return;
+	// import html as dom and export is as xml to avoid xml malformations
+	let dom = xmlparser.parseFromString(html, "text/html");
```

Comment:
This won't work. These fields aren't HTML and can't be parsed as HTML. They just use a subset of HTML for specific rich-text formatting. I'm not totally clear what this PR is doing, but if the goal is to treat Zotero fields as unrestricted HTML, it's not valid and should be closed.

`replaceFormatting()` was technically already wrong here. Zotero's citation processor ignores unmatched tags, so a title `<b> is not <strong>` is treated literally without interpreting as rich text. In other words, single regexes should be matching ungreedily on each opening and closing tag and capturing the value in between, instead of there being separate regexes for the opening and closing tags. It would be good to fix this, though it probably doesn't come up much.

But what you have is much worse, because it's treating the string as full HTML. So `<b> is not <strong>` would get converted to:

```xml
<title level="a">
	<hi rend="bold"> is not <strong/>
	</hi>
</title>
```

which we obviously don't want.

Everything that's not the specific supported matching tags should be encoded as plain text. If that's what the translator was doing before, it was already correct (other than replacing non-matching tags).

Reply:
Hi Dan,

«This won't work.» It works on some hundreds of real life records, for example on [this group](https://www.zotero.org/groups/5018778/rougemont/library), but not on your case, you are right.

Users with brackets in their titles will have some hand work, like for all TEI edition (ex: italic in a title is usually a title, Zotero can’t infer it, a TEI editor should), time is saved for most common cases. That said… if TEI.js is reworked, let’s try to do the best.

Have you some clue to find the code doing this nice job in Zotero desktop ?
![image](https://github.com/zotero/translators/assets/5686231/0924ae72-435e-47b4-b708-3da5930bd7d2)

The RDF solution is not an option for TEI. This is not semantic tagging. The original author of TEI.js, (C) 2010 Stefan Majewski, was right from a TEI point of view, even if his solution is not yet perfect.
`Some people put &lt;b&gt;bold&lt;/b&gt; in &lt;i&gt;titles&lt;/i&gt; ?`

Reply:
> «This won't work.» It works on some hundreds of real life records, for example on [this group](https://www.zotero.org/groups/5018778/rougemont/library), but not on your case, you are right.

I don't mean "won't work" as in it won't produce the correct result in many cases. I mean it doesn't work conceptually. It's not an HTML field and therefore can't be treated like it is.

> Have you some clue to find the code doing this nice job in Zotero desktop ?

https://github.com/zotero/zotero/blob/04496fadcd73ad7db85d7a9c662fcbffac80d72e/chrome/content/zotero/itemTree.jsx#L2570-L2658

Reply:
NICE ! Thanks. Because it is documented as “Private methods”, should I guess the good approach is to acclimate such code in TEI.js ? It seems that this PR needs a full rewrite. What should I do ? Submit another PR when work is done ?

Reply:
Nothing to do with "private" — that code just isn't remotely related to the translator environment. (Most zotero/zotero code has no connection to translators.)

Sure, probably easiest to close this and open a new one when want us to look at something new. Thanks!

Reply:
Hi @dstillman,
I had to continue other works on TEI.js (other features features like indentation, or more fields supported especially some extra used by APA7…). I have now acclimate the piece of code you point to me. No more regex are going through html.
https://github.com/glorieux-f/translators/blob/tei_export/TEI.js#L140
May I submit this new version ? It’s quite a full rewrite of previous version (~10 years). Feel free to ask for more upgrades to conform the code with present Zotero standards.

Reply:
Sure!

Reply:
There is maybe a place where I’m not perfect, parsing the Extra field. You probably know where is the piece of code doing this good job before sending to Style. 
https://github.com/glorieux-f/translators/blob/tei_export/TEI.js#L307
Other thing, I don’t know how to work well with Pull Request. Isn’t it your responsibility to close it ?

Reply:
You can certainly always close your own PRs, but I'll close this one.

We can talk about Extra on the new PR.